src/libs/OnyxAwareParser.ts

```diff
@@ -0,0 +1,42 @@
+import ExpensiMark from 'expensify-common/lib/ExpensiMark';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '@src/ONYXKEYS';
+
+const parser = new ExpensiMark();
+
+const reportIDToNameMap: Record<string, string> = {};
+const accountIDToNameMap: Record<string, string> = {};
+
+Onyx.connect({
+    key: ONYXKEYS.COLLECTION.REPORT,
+    callback: (report) => {
+        if (!report) {
+            return;
+        }
+
+        reportIDToNameMap[report.reportID] = report.reportName ?? report.displayName ?? report.reportID;
+    },
+});
+
+Onyx.connect({
+    key: ONYXKEYS.PERSONAL_DETAILS_LIST,
+    callback: (personalDetailsList) => {
+        Object.values(personalDetailsList ?? {}).forEach((personalDetails) => {
+            if (!personalDetails) {
+                return;
+            }
+
+            accountIDToNameMap[personalDetails.accountID] = personalDetails.displayName ?? String(personalDetails.accountID);
```

Comment:
same here

Reply:
Is it better to fallback to `personalDetails.login` here? On the profile page, we also fallback to show login email if display name is empty
<img width="625" alt="image" src="https://github.com/Expensify/App/assets/117511920/c3fd3ad5-7050-44e2-9094-69459a82228c">

Reply:
Actually, we should use the `personalDetails.login` instead of `.displayName` because if we use `.displayName` we won't be able to return from it to the id-based HTML mention tag. Will update it